src/elm/View/Form/Input.elm

```diff
@@ -0,0 +1,97 @@
+module View.Form.Input exposing (init, input, toHtml, withCounter)
+
+import Html exposing (Html, div, input, li, text, ul)
+import Html.Attributes exposing (class, disabled, id, placeholder, value)
+import Html.Events exposing (onInput)
+import Session.Shared exposing (Translators)
+import View.Form
+
+
+{-| Required options for an input
+-}
+type alias InputOptions a =
+    { label : String
+    , id : String
+    , onInput : String -> a
+    , disabled : Bool
+    , value : String
+    , placeholder : Maybe String
+    , problems : Maybe (List String)
+    , translators : Translators
+    , maximumCounterValue : Maybe Int
+    }
+
+
+type alias RequiredInputOptions a =
+    { label : String
+    , id : String
+    , onInput : String -> a
+    , disabled : Bool
+    , value : String
+    , placeholder : Maybe String
+    , problems : Maybe (List String)
+    , translators : Translators
+    }
+
+
+init : RequiredInputOptions a -> InputOptions a
+init options =
+    { label = options.label
+    , id = options.id
+    , onInput = options.onInput
+    , disabled = options.disabled
+    , value = options.value
+    , placeholder = options.placeholder
+    , problems = options.problems
+    , maximumCounterValue = Nothing
+    , translators = options.translators
+    }
+
+
+toHtml : InputOptions a -> Html a
+toHtml options =
+    div [ class "mb-10 relative" ]
+        [ View.Form.label options.id options.label
+        , input options
+        , case options.maximumCounterValue of
+            Just number ->
+                div [ class "input-label pr-1 text-right text-purple-100 font-bold mt-1 absolute right-0" ]
+                    [ text <|
+                        options.translators.tr
+                            "edit.input_counter"
+                            [ ( "current", String.fromInt <| String.length options.value )
+                            , ( "max", String.fromInt number )
+                            ]
+                    ]
+
+            Nothing ->
+                text ""
```

Comment:
I suggest moving the counter into a separate module (it could be used also for textareas) and use it like this:

```elm
InputCounter.view tr 12 form.username
```
The module itself:
```elm
-- src/elm/View/Form/InputCounter.elm
module View.Form.InputCounter exposing (..)

import Html exposing (Html, div, text)
import Html.Attributes exposing (class)
import I18Next


view : (String -> I18Next.Replacements -> String) -> Int -> String -> Html msg
view tr max str =
    div [ class "input-counter" ]
        [ text <|
            tr "edit.input_counter"
                [ ( "current", String.fromInt <| String.length str )
                , ( "max", String.fromInt max )
                ]
        ]
```